### ISSUE TITLE
How do I align space-separated values?

### ISSUE THREAD

**Issue by airblade** [tommcdo/vim-lion#12]
Often I find myself wanting to align columns which are only separated by spaces.  For example:

```ruby
# from a chef cookbook_file resource
source 'https://github.com/leev/ngx_http_geoip2_module/archive/1.0.tar.gz'
checksum '0af7668f5c0baed8ab6c5fe0a4ed8eb8a76e23383db69dc31d25ea2c7a785262' 
```

I would like to turn this into:

```ruby
source   'https://github.com/leev/ngx_http_geoip2_module/archive/1.0.tar.gz'
checksum '0af7668f5c0baed8ab6c5fe0a4ed8eb8a76e23383db69dc31d25ea2c7a785262'
```

However I can't quite figure out how to achieve this with vim-lion.  Any pointers please?

P.S. Thanks for a terrific plugin.

**Comment by tommcdo** [comment]
For this particular case, you could use `'` as your alignment character, e.g. `glip'`. However, I suspect you're looking for a more general solution.

With no leading indentation, using a space as your alignment character works. But when there is leading indentation made up of spaces, it doesn't, as it basically aligns some stuff that's already aligned.

If you type `/` as your alignment character, you're prompted to enter a pattern (or press Enter to accept the default, a literal `/`). I played around a bit and found that the pattern `\S\zs\s` seems to work when using the `gl` operator (add spaces to the left of the alignment character) for this need. Effectively, that's a non-whitespace character (`\S`) followed by a whitespace character (`\s`). The `\zs` anchor indicates the start of the match, so the non-whitespace will not be included in the match. This means no spaces will be inserted before the non-whitespace character, leaving the spaces to be inserted only before the matching whitespace.

**TL;DR** `glip/\S\zs\s<CR>`

**Comment by airblade** [comment]
Thanks for the detailed explanation.  Your solution works perfectly :)

There's no way I'll remember it though so I'll come up with some kind of map in my vimrc!

**Comment by tommcdo** [comment]
I might make a special case for `<Space>` as an alignment character. I'm not sure the natural behaviour is useful in general.

**Comment by tommcdo** [comment]
In fact, I will definitely do this. In its present state, `<Space>` as an alignment character is totally useless. It doesn't behave well with indentation, it's garbage for counts, ...

Reopening this issue as a means of tracking the feature.

**Comment by airblade** [comment]
That's a neat solution, thank you.  Makes my life easier :)
